server/session.go

```diff
@@ -328,8 +353,14 @@ func (s *Session) cleanUp(expired bool) {
 
 // Message received, convert bytes to ClientComMessage and dispatch
 func (s *Session) dispatchRaw(raw []byte) {
+	now := time.Now().UTC().Round(time.Millisecond)
 	var msg ClientComMessage
 
+	if s.terminating {
+		log.Println("s.dispatch: message received on a terminating session", s.sid)
+		s.queueOut(ErrPermissionDenied("", "", now))
```

Comment:
I think it should be one of 5xx errors. 4xx means the client did something wrong. 5xx means the client is fine while the server cannot handle a correct request from a client.

We have `ErrLockedReply` as 423. Consider redefining it as 503 and useing here and in case of a locked topic.

Reply:
Okay, changed ErrLocked to be a 503.